cita-executor/core/src/trie_db.rs

```diff
@@ -0,0 +1,102 @@
+use std::sync::Arc;
+
+use parking_lot::RwLock;
+use std::collections::HashMap;
+use cita_database::{DataCategory, Database};
+use cita_database::error::DatabaseError;
+
+#[derive(Debug)]
+pub struct TrieDB<DB>
+where
+    DB: Database,
+{
+    db:    Arc<DB>,
+    cache: Arc<RwLock<HashMap<Vec<u8>, Vec<u8>>>>,
+}
+
+impl<DB> TrieDB<DB>
+where
+    DB: Database,
+{
+    pub fn new(db: Arc<DB>) -> Self {
+        TrieDB {
+            db,
+            cache: Arc::new(RwLock::new(HashMap::new())),
+        }
+    }
+}
+
+/// "TrieDB" provides state read/write capabilities for executor.
+impl<DB> cita_trie::DB for TrieDB<DB>
+where
+    DB: Database,
+{
+    type Error = DatabaseError;
+
+    fn get(&self, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error> {
+        match self.cache.read().get(key) {
+            Some(v) => Ok(Some(v.to_vec())),
+            None => self.db.get(Some(DataCategory::State), key),
+        }
+    }
+
+    fn insert(&self, key: Vec<u8>, value: Vec<u8>) -> Result<(), Self::Error> {
+        self.cache.write().insert(key, value);
+        Ok(())
+    }
+
+    fn contains(&self, key: &[u8]) -> Result<bool, Self::Error> {
+        if self.cache.read().contains_key(key) {
+            Ok(true)
+        } else {
+            self.db.contains(Some(DataCategory::State), key)
+        }
+    }
+
+    fn remove(&self, _key: &[u8]) -> Result<(), Self::Error> {
+        Ok(())
+    }
+
+    fn insert_batch(&self, keys: Vec<Vec<u8>>, values: Vec<Vec<u8>>) -> Result<(), Self::Error> {
+        let mut cache = self.cache.write();
+        for i in 0..keys.len() {
+            let key = keys[i].clone();
+            let value = values[i].clone();
+            cache.insert(key, value);
+        }
+        Ok(())
+    }
+
+    fn remove_batch(&self, _keys: &[Vec<u8>]) -> Result<(), Self::Error> {
+        Ok(())
+    }
+
```

Comment:
Not implement yet? Together with `remove()`

Reply:
...
I'll use `unimplemented`.

Reply:
Yes, `unimplemented` is much better.

Reply:
DONE.
It's the cods from old project...